pilot/proxy/envoy/config.go

```diff
@@ -542,10 +543,13 @@ func buildDestinationHTTPRoutes(service *model.Service,
 		useDefaultRoute := true
 		rules := config.RouteRules(instances, service.Hostname)
 		// sort for output uniqueness
+		// if v1alpha2 rules are returned, len(rules) <= 1 is guaranteed
+		// because v1alpha2 rules are unique per host.
 		model.SortRouteRules(rules)
+
```

Comment:
Add a TODO to eliminate the for loop below since we now have only a single rule per destination.. Alternatively, we can fold the buildHTTPRoute logic under this function..